### PR TITLE
docs(reference/reply): When using async-await, need return

### PR DESCRIPTION
**Pull request by radiorz**
…y object to signal fastify wait for your further response.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by radiorz** [comment]
#4412

**Review by mcollina — APPROVED** [review]
lgtm

File: docs/Reference/Reply.md [hunk 1]
```diff
@@ -689,6 +699,17 @@ fastify.get('/streams', function (request, reply) {
 })
 ```
 
+When using async-await, please return the reply object to signal fastify wait 
+for your further response.
+```js
+const fs = require('fs')
+fastify.get('/streams', function (request, reply) {
```
Comment (Fdawgs) [inline docs/Reference/Reply.md:706]:
```suggestion
fastify.get('/streams', async function (request, reply) {
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]
Doesn't read massively well, needs rethinking.

File: docs/Reference/Reply.md [hunk 2]
```diff
@@ -674,6 +674,16 @@ fastify.get('/streams', function (request, reply) {
   reply.send(stream)
 })
 ```
+When using async-await, please return the reply object to signal fastify wait 
+for your further response.
```
Comment (Fdawgs) [inline docs/Reference/Reply.md:678]:
```suggestion
When using async-await you will need to return the reply object:
```

File: docs/Reference/Reply.md [hunk 3]
```diff
@@ -689,6 +699,17 @@ fastify.get('/streams', function (request, reply) {
 })
 ```
 
+When using async-await, please return the reply object to signal fastify wait 
+for your further response.
```
Comment (Fdawgs) [inline docs/Reference/Reply.md:703]:
```suggestion
When using async-await you will need to return the reply object:
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]
Cut down on formalities and fluff.

**Comment by L2jLiga** [comment]
I feel like we have 2 almost same sections in [`Routes.md`](https://github.com/sergejostir/fastify/blob/da6162f6290aac5a051c47355c824f9c9bdde2b9/docs/Routes.md#async-await) and in `Reply.md` covering the same topic

Also in `Routes.md` stated that you can just await reply, however in this docs update is stated that *you will need to return the reply object* which confuse me 😅

**Comment by climba03003** [comment]
> I feel like we have 2 almost same sections in [`Routes.md`](https://github.com/sergejostir/fastify/blob/da6162f6290aac5a051c47355c824f9c9bdde2b9/docs/Routes.md#async-await) and in `Reply.md` covering the same topic
> 
> Also in `Routes.md` stated that you can just await reply, however in this docs update is stated that _you will need to return the reply object_ which confuse me 😅

Both of them should be the same, when you return `reply` the await job is done by `fastify`. When you `await` yourself, it is done explicitly by the developer.

File: docs/Reference/Reply.md [hunk 4]
```diff
@@ -674,6 +674,15 @@ fastify.get('/streams', function (request, reply) {
   reply.send(stream)
 })
 ```
+When using async-await you will need to return the reply object:
```
Comment (L2jLiga) [inline docs/Reference/Reply.md:677]: @climba03003, both options returns Promise, the difference only is what inside - nothing or reply object, both of them handled same way and this part confuses me a bit - why I need to return reply if I can just await it?

```suggestion
When using async-await you will need to return or await the reply object:
```

Comment (climba03003) [reply docs/Reference/Reply.md:677]: The difference is just `implicit` or `explicit` handle of `Promise`.
`return` or `await` both `await`ing `reply` in the end.

The `await` here is used to signal the `payload` is not ready yet if you do something off the track.

**Review by L2jLiga — APPROVED** [review]
LGTM

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**